### DISCUSSION TITLE
Regarding the last pull request

### DISCUSSION THREAD

**Discussion by AntoineRestivo** [benoitseron/BosonSampling.jl#46] [General]
https://github.com/benoitseron/BosonSampling.jl/blob/3aef4574def323fa585b6246c7b4fe0926c78b97/src/example_usage.jl#L24

That's exactly what I want to avoid, the number of photons is repeated 3 times ...

**Comment by benoitseron** [comment]
struct Input

function Input{everything}

function Input{RandomGramMatrix}(r::ModeOccupation)
new(Input{PartDist}(r, GramMatrix{PartDist}(r.n, rand_gram_matrix(r.n))) )
